In [2]:
from sklearn.datasets import fetch_openml
mnist = fetch_openml('mnist_784', version=1)
mnist.keys()


dict_keys(['data', 'target', 'frame', 'categories', 'feature_names', 'target_names', 'DESCR', 'details', 'url'])

inspect the arrays are in mnist, both data and target

A data key containing an array with one row per instance and one column per
feature
• A target key containing an array with the labels

In [26]:
X, y = mnist["data"], mnist["target"]

print(X.shape)
print(y.shape)


print(X[0])

(70000, 784)
(70000,)


KeyError: 0

we see that X has a shape of 70000 rows and 784 columns, ie. features. 
y on the other hand has only 70000 observations. perfect. 

Theres 784 features because there are 28x28 pixels

If you want to look at one digit from the dataset, you grab an instance's feature vector (ie a whole row, where the it's a vector cuz it's a 1X784), reshape it to a 28X28 array and display it using matplotlibs imshow() function. 

In [21]:
import matplotlib as mpl 
import matplotlib.pyplot as plt

some_digit = X[0] #first row, first feature vector
some_digit_image = some_digit.reshape(28, 28) 
plt.imshow(some_digit_image, cmap = mpl.cm.binary, interpolation= "nearest") # interpolation shades blends the edges of 
# pixels together to make them less harsh
plt.axis("off")
plt.show()

KeyError: 0